impl/src/main/java/com/sun/faces/spi/ServiceFactoryUtils.java

```diff
@@ -143,4 +149,29 @@ static String[] getServiceEntries(String key) {
 
     }
 
+    /**
+     * Cleans line in service file by processing comments and trimming.
+     *
+     * @param line
+     *         Line to clean.
+     * @return The cleaned line, if it isn't empty after removing the comment and trimming,
+     *         otherwise <code>null</code>.
+     */
+    private static String cleanupServiceLine(String line) {
+        // Process comments, starting at '#'.
+        int commentIndex = line.indexOf('#');
+        if (commentIndex == 0) {
+            // The whole line is a comment, no need to process it.
+            return null;
+        } else if (commentIndex > 0) {
+            // Cut everything of after '#'.
+            line = line.substring(0, commentIndex);
+        }
+        line = line.trim();
+        if (line.isEmpty()) {
+            return null;
+        } else {
+            return line;
+        }
```

Comment:
Here is what MyFaces does it might look a little cleaner...

```java
                                int idx = serviceImplName.indexOf('#');
                                if (idx >= 0)
                                {
                                    serviceImplName = serviceImplName.substring(0, idx);
                                }
                                serviceImplName = serviceImplName.trim();

                                if (serviceImplName.length() != 0)
                                {
                                    results.add(serviceImplName);
                                }
```

Reply:
I appreciate your effort, but MyFaces is under the Apache 2 License, which is to my understanding not really compatible with the EPL. I don't want to copy their code (even if they probably don't care).

Reply:
You don't need to quite copy it but you can see its a much simpler version.  And Apache 2 is basically the most friendly open source license there is...